backend/controllers/googlegroup.controller.js

```diff
@@ -0,0 +1,88 @@
+const GoogleGroup = require('../models/googlegroup.model')
+
+const getAllGoogleGroups = async () => {
+    try {
+        let result = []
+        const allPairs = await GoogleGroup.find()
+        allPairs.map((pair) => result.push(pair))
+        console.log('Fetched all google group pairs')
+        return result
+    } catch (err) {
+        console.log('Error: ' + err)
+    }
+}
+
+const getGoogleGroup = async (req, res) => {
+    const { field, useEmail } = req.body
+    try {
+        let userGroup = {}
+        if (useEmail) {
+            userGroup = await GoogleGroup.findOne({ email: field })
+        } else {
+            userGroup = await GoogleGroup.findOne({ watiam: field })
+        }
+        res.status(200).json(userGroup)
+        return userGroup
+    } catch (err) {
+        res.status(400).json('Error: ' + err)
+    }
+}
+
+const updateGoogleGroups = async (req, res) => {
+    const newPairs = req.body
+
+    const currentPairs = await getAllGoogleGroups()
+
+    const currentEmails = currentPairs.map((pair) => pair.email)
+    const newEmails = newPairs.map((pair) => pair.email)
+
+    const emailsToDelete = currentEmails.filter(
```

Comment:
make deleteEmails a new function and call it within this one. similar to update/add so we have a separation of logic there.